crates/red_knot_python_semantic/src/types.rs

```diff
@@ -318,6 +318,9 @@ pub struct FunctionType<'db> {
 
     /// types of all decorators on this function
     decorators: Vec<Type<'db>>,
+
+    /// return type for this function
```

Comment:
nit: the function will always have a _return type_, even if it doesn't have a _return annotation_. If it doesn't have a return annotation, we will either just infer the function's return type as `Unknown` (what mypy does), or analyse the body of the function to figure out the return type (what pyright does)

```suggestion
    /// type inferred from this function's return annotation
    /// (if it has a return annotation)
```

Reply:
I agree with this nit, but I have a nit on the nit: although we use the term "infer" pretty broadly, I wouldn't choose to use that term for cases where we are simply resolving an annotation expression to a type:
```suggestion
    /// annotated return type for this function, if any
```

Reply:
DOUBLE NIT. I love it. 😁